projects/core/src/occ/config-loader/converters/occ-loaded-config-converter.ts

```diff
@@ -1,23 +1,59 @@
 import { Injectable } from '@angular/core';
-import { I18nConfig } from '../../i18n';
-import { BaseSite } from '../../model/misc.model';
-import { SiteContextConfig } from '../../site-context/config/site-context-config';
+import { I18nConfig } from '../../../i18n';
+import { BaseSite } from '../../../model/misc.model';
+import { SiteContextConfig } from '../../../site-context/config/site-context-config';
 import {
   BASE_SITE_CONTEXT_ID,
   CURRENCY_CONTEXT_ID,
   LANGUAGE_CONTEXT_ID,
   THEME_CONTEXT_ID,
-} from '../../site-context/providers/context-ids';
-import { Occ } from '../occ-models/occ.models';
+} from '../../../site-context/providers/context-ids';
+import { Converter } from '../../../util/converter.service';
+import { Occ } from '../../occ-models/occ.models';
+import { OccLoadedConfig } from '../occ-loaded-config';
 import { JavaRegExpConverter } from './java-reg-exp-converter';
-import { OccLoadedConfig } from './occ-loaded-config';
 
 @Injectable({ providedIn: 'root' })
-export class OccLoadedConfigConverter {
+export class OccLoadedConfigConverter
+  implements Converter<BaseSite, OccLoadedConfig> {
+  // TODO: remove javaRegExpConverter from constructor in 4.0
   constructor(private javaRegExpConverter: JavaRegExpConverter) {}
 
-  fromOccBaseSites(baseSites: BaseSite[], currentUrl: string): OccLoadedConfig {
-    const baseSite = baseSites.find((site) =>
+  convert(source: BaseSite, target?: OccLoadedConfig): OccLoadedConfig {
+    // Although `stores` property is an array, typically there is only one store.
+    // So we return the first store from the list.
+    const baseStore = source.stores && source.stores[0];
+    if (!baseStore) {
+      throw this.getError(
+        `Current base site (${source.uid}) doesn't have any base store.`
+      );
+    }
```

Comment:
Would you mind moving this logic to the standard `BaseSiteNormalizer`? 

Let's add to the UI model of BaseSite a new property `.baseStore` and in `BaseSiteNormalizer` class let's populate it:
```ts
// Although `stores` property is an array, typically there is only one store.
// So we take the first store from the list.
target.baseStore = source?.stores?.[0];
delete source?.stores;
```

Thanks to that we'll avoid that conversion logic in lower layers.